lib/chef/provider/package.rb

```diff
@@ -490,6 +490,38 @@ def allow_downgrade
           false
         end
       end
+
+      # Set provider priority
+      require 'chef/chef_class'
+      require 'chef/provider/package/dpkg'
+      require 'chef/provider/package/homebrew'
+      require 'chef/provider/package/macports'
+      require 'chef/provider/package/apt'
+      require 'chef/provider/package/yum'
+      require 'chef/provider/package/zypper'
+      require 'chef/provider/package/portage'
+      require 'chef/provider/package/pacman'
+      require 'chef/provider/package/ips'
+      require 'chef/provider/package/solaris'
+      require 'chef/provider/package/smartos'
+      require 'chef/provider/package/aix'
+      require 'chef/provider/package/paludis'
+
+      Chef.set_provider_priority_array :package, [ Homebrew, Macports ], os: "darwin"
+
+      Chef.set_provider_priority_array :package,  Dpkg,     os: "linux"
```

Comment:
dpkg isn't a default for all linuxen, this should only be `platform_family: "debian"`